connection_test.go

```diff
@@ -437,3 +437,38 @@ func TestGetInvalidQueryStatus(t *testing.T) {
 		t.Error("expected an error")
 	}
 }
+
+func TestExecWithServerSideError(t *testing.T) {
+	postQueryMock := func(_ context.Context, _ *snowflakeRestful,
```

Comment:
while mock is ok to test when response is empty, do we know what is the possible reason of this response?